### ISSUE TITLE
Discovering firefoxpkg_intl but installing firefoxpkgintl

### ISSUE THREAD

**Issue by Apfelpom** [robjschroeder/App-Auto-Patch#13]
We have excluded all other Firefox-Labels in the ignoreddLabel to match the one and only firefoxpkg_intl: 
`ignoredLabels="${7:="firefox firefox_da firefox_intl firefoxdeveloperedition firefoxesr firefoxesr_intl firefoxpkg"}"`
_(BTW: there is a typo in 1.0.14 for firefoxdeveloperedition and we don't need firefoxesrpkg as it's the alternate name of firefoxesr)._

During the discovery the label will be recognized:
`[NOTICE] Queueing firefoxpkg_intl`
But when coming to installation, the label lost it's underscore:
```
App Auto-Patch: Installing firefoxpkgintl...
App Auto-Patch: [INFO] Update swiftDialog: progress: increment 
REQ   : firefoxpkgintl : ################## Start Installomator v. 10.5, date 2023-10-16
INFO  : firefoxpkgintl : ################## Version: 10.5
INFO  : firefoxpkgintl : ################## Date: 2023-10-16
INFO  : firefoxpkgintl : ################## firefoxpkgintl
INFO  : firefoxpkgintl : Installomator did not close any apps, so no need to reopen any apps.
ERROR : firefoxpkgintl : ERROR: unknown label firefoxpkgintl
REQ   : firefoxpkgintl : ################## End Installomator, exit code 1 
```

**Comment by robjschroeder** [comment]
firefoxpkg_intl seems to be working correctly now. 

`App Auto-Patch: 2023-10-24 09:21:38 - [INFO] Installing firefoxpkg_intl...
App Auto-Patch: 2023-10-24 09:21:38 - [INFO] Update swiftDialog: progress: increment 11
App Auto-Patch: 2023-10-24 09:21:39 - [INFO] Update swiftDialog: icon: /Applications/Firefox.app
App Auto-Patch: 2023-10-24 09:21:39 - [INFO] Update swiftDialog: progresstext: Checking Firefox …
App Auto-Patch: 2023-10-24 09:21:39 - [INFO] Update swiftDialog: listitem: index: 1, status: wait, statustext: Checking …
2023-10-24 09:21:39 : REQ   : firefoxpkg_intl : ################## Start Installomator v. 10.5, date 2023-10-15
2023-10-24 09:21:39 : INFO  : firefoxpkg_intl : ################## Version: 10.5
2023-10-24 09:21:39 : INFO  : firefoxpkg_intl : ################## Date: 2023-10-15
2023-10-24 09:21:39 : INFO  : firefoxpkg_intl : ################## firefoxpkg_intl
2023-10-24 09:21:39 : WARN  : firefoxpkg_intl : Found language en-US to be used for Firefox.
2023-10-24 09:21:39 : WARN  : firefoxpkg_intl : Using language en-US for download.
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : setting variable from argument BLOCKING_PROCESS_ACTION=prompt_user NOTIFY=silent LOGO=appstore 
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : setting variable from argument DIALOG_CMD_FILE="/var/tmp/dialog.appAutoPatch.fTle7"
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : setting variable from argument DIALOG_LIST_ITEM_NAME='Firefox'
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : BLOCKING_PROCESS_ACTION=prompt_user
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : NOTIFY=silent
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : LOGGING=INFO
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : Label type: pkg
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : archiveName: Firefox.pkg
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : App(s) found: /Applications/Firefox.app
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : found app at /Applications/Firefox.app, version 119.0, on versionKey CFBundleShortVersionString
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : appversion: 119.0
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : Latest version of Firefox is 119.0
2023-10-24 09:21:40 : INFO  : firefoxpkg_intl : There is no newer version available.
2023-10-24 09:21:42 : INFO  : firefoxpkg_intl : Installomator did not close any apps, so no need to reopen any apps.
2023-10-24 09:21:42 : REQ   : firefoxpkg_intl : No newer version.
2023-10-24 09:21:42 : REQ   : firefoxpkg_intl : ################## End Installomator, exit code 0 `

**Comment by Apfelpom** [comment]
I've tested with ignoring Firefox labels one by one, and ignoring them with a wildcard and add the firefoxpkg_intl to the required apps. In both cases version 2.0.0b8 failed.

```
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] Ignoring label firefox.
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] Ignoring label firefox_da.
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] Ignoring label firefox_intl.
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] Ignoring label firefoxdeveloperedition.
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] Ignoring label firefoxesr.
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] Ignoring label firefoxesr_intl.
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] Ignoring label firefoxpkg.
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] Searching for Firefox.app
App Auto-Patch: 2023-10-24 17:25:56 - [INFO] Found Firefox.app version 118.0.2
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] Label: firefoxpkg_intl
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] --- found app at /Applications/Firefox.app
App Auto-Patch: 2023-10-24 17:25:56 - [NOTICE] Verifying: /Applications/Firefox.app
App Auto-Patch: 2023-10-24 17:25:56 - [INFO] Update swiftDialog: progresstext: Verifying /Applications/Firefox.app
id: : no such user
posix_spawn(): 2: No such file or directory
App Auto-Patch: 2023-10-24 17:26:03 - [NOTICE] --- Installed version: 118.0.2
App Auto-Patch: 2023-10-24 17:26:03 - [NOTICE] --- Newest version: 2023-10-24 17:26:02 : WARN  :  : Found language  to be used for Firefox.
2023-10-24 17:26:02 : WARN  :  : Using language  for download.
119.0
App Auto-Patch: 2023-10-24 17:26:03 - [NOTICE] Queueing firefoxpkg_intl
App Auto-Patch: 2023-10-24 17:26:03 - [INFO] 1password8 apparency cloudya daisydisk firefoxpkg_intl
...
App Auto-Patch: 2023-10-24 17:27:46 - [NOTICE] Labels to install:  1password8 apparency cloudya daisydisk firefoxpkg_intl firefoxpkgintl jamfmigrator mactracker packages suspiciouspackage theunarchiver
...
App Auto-Patch: 2023-10-24 17:28:44 - [INFO] Installing firefoxpkgintl...
App Auto-Patch: 2023-10-24 17:28:44 - [INFO] Update swiftDialog: progress: increment 7
App Auto-Patch: 2023-10-24 17:28:45 - [INFO] Update swiftDialog: icon: /Applications/.app
App Auto-Patch: 2023-10-24 17:28:45 - [INFO] Update swiftDialog: progresstext: Checking  …
App Auto-Patch: 2023-10-24 17:28:45 - [INFO] Update swiftDialog: listitem: index: 5, status: wait, statustext: Checking …
2023-10-24 17:28:45 : REQ   : firefoxpkgintl : ################## Start Installomator v. 10.5, date 2023-10-16
2023-10-24 17:28:45 : INFO  : firefoxpkgintl : ################## Version: 10.5
2023-10-24 17:28:45 : INFO  : firefoxpkgintl : ################## Date: 2023-10-16
2023-10-24 17:28:45 : INFO  : firefoxpkgintl : ################## firefoxpkgintl
2023-10-24 17:28:45 : INFO  : firefoxpkgintl : Installomator did not close any apps, so no need to reopen any apps.
2023-10-24 17:28:45 : ERROR : firefoxpkgintl : ERROR: unknown label firefoxpkgintl
2023-10-24 17:28:45 : REQ   : firefoxpkgintl : ################## End Installomator, exit code 1 
```

Ignoring firefox* labels:

```
App Auto-Patch: 2023-10-24 17:54:27 - [NOTICE] Ignoring all lables with firefox*
App Auto-Patch: 2023-10-24 17:54:27 - [INFO] Writing ignored label firefox_intl to configuration plist
App Auto-Patch: 2023-10-24 17:54:27 - [INFO] Writing ignored label firefoxesr to configuration plist
App Auto-Patch: 2023-10-24 17:54:27 - [INFO] Writing ignored label firefoxpkg_intl to configuration plist
App Auto-Patch: 2023-10-24 17:54:27 - [INFO] Writing ignored label firefox_da to configuration plist
App Auto-Patch: 2023-10-24 17:54:27 - [INFO] Writing ignored label firefoxesr_intl to configuration plist
App Auto-Patch: 2023-10-24 17:54:27 - [INFO] Writing ignored label firefoxpkg to configuration plist
App Auto-Patch: 2023-10-24 17:54:28 - [INFO] Writing ignored label firefox to configuration plist
App Auto-Patch: 2023-10-24 17:54:28 - [INFO] Writing ignored label firefoxdeveloperedition to configuration plist
App Auto-Patch: 2023-10-24 17:54:28 - [NOTICE] Attempting to populate required labels
App Auto-Patch: 2023-10-24 17:54:28 - [INFO] Writing required label firefoxpkg_intl to configuration plist

App Auto-Patch: 2023-10-24 17:56:41 - [NOTICE] Labels to install:  1password8 apparency cloudya daisydisk firefoxpkg_intl firefoxpkgintl jamfmigrator mactracker packages suspiciouspackage theunarchiver 


App Auto-Patch: 2023-10-24 17:57:27 - [INFO] Installing firefoxpkg_intl...
App Auto-Patch: 2023-10-24 17:57:27 - [INFO] Update swiftDialog: progress: increment 7
App Auto-Patch: 2023-10-24 17:57:28 - [INFO] Update swiftDialog: icon: /Applications/Firefox.app
App Auto-Patch: 2023-10-24 17:57:28 - [INFO] Update swiftDialog: progresstext: Checking Firefox …
App Auto-Patch: 2023-10-24 17:57:28 - [INFO] Update swiftDialog: listitem: index: 4, status: wait, statustext: Checking …
2023-10-24 17:57:28 : REQ   : firefoxpkg_intl : ################## Start Installomator v. 10.5, date 2023-10-15
2023-10-24 17:57:28 : INFO  : firefoxpkg_intl : ################## Version: 10.5
2023-10-24 17:57:28 : INFO  : firefoxpkg_intl : ################## Date: 2023-10-15
2023-10-24 17:57:28 : INFO  : firefoxpkg_intl : ################## firefoxpkg_intl
2023-10-24 17:57:28 : WARN  : firefoxpkg_intl : Found language de-DE to be used for Firefox.
2023-10-24 17:57:28 : INFO  : firefoxpkg_intl : No locale matching 'de-DE', trying 'de'
2023-10-24 17:57:28 : WARN  : firefoxpkg_intl : Using language de for download.
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : setting variable from argument BLOCKING_PROCESS_ACTION=prompt_user NOTIFY=silent LOGO=aundo INTERRUPT_DND=yes DEBUG=2 
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : setting variable from argument DIALOG_CMD_FILE="/var/tmp/dialog.appAutoPatch.ACGpd"
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : setting variable from argument DIALOG_LIST_ITEM_NAME='Firefox'
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : BLOCKING_PROCESS_ACTION=prompt_user
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : NOTIFY=silent
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : LOGGING=INFO
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : LOGO=/System/Applications/App Store.app/Contents/Resources/AppIcon.icns
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : Label type: pkg
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : archiveName: Firefox.pkg
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : App(s) found: /Applications/Firefox.app
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : found app at /Applications/Firefox.app, version 118.0.2, on versionKey CFBundleShortVersionString
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : appversion: 118.0.2
2023-10-24 17:57:29 : INFO  : firefoxpkg_intl : Latest version of Firefox is 119.0
2023-10-24 17:57:30 : REQ   : firefoxpkg_intl : Downloading https://download.mozilla.org/?product=firefox-pkg-latest-ssl&os=osx&lang=de to Firefox.pkg
2023-10-24 17:57:51 : REQ   : firefoxpkg_intl : no more blocking processes, continue with update
2023-10-24 17:57:51 : REQ   : firefoxpkg_intl : Installing Firefox
2023-10-24 17:57:51 : INFO  : firefoxpkg_intl : Verifying: Firefox.pkg
2023-10-24 17:57:51 : INFO  : firefoxpkg_intl : Team ID: 43AQ936H96 (expected: 43AQ936H96 )
2023-10-24 17:57:51 : INFO  : firefoxpkg_intl : Installomator did not close any apps, so no need to reopen any apps.
2023-10-24 17:57:51 : REQ   : firefoxpkg_intl : ################## End Installomator, exit code 0 
...
App Auto-Patch: 2023-10-24 17:57:51 - [INFO] Installing firefoxpkgintl...
App Auto-Patch: 2023-10-24 17:57:51 - [INFO] Update swiftDialog: progress: increment 7
App Auto-Patch: 2023-10-24 17:57:52 - [INFO] Update swiftDialog: icon: /Applications/.app
App Auto-Patch: 2023-10-24 17:57:52 - [INFO] Update swiftDialog: progresstext: Checking  …
App Auto-Patch: 2023-10-24 17:57:52 - [INFO] Update swiftDialog: listitem: index: 5, status: wait, statustext: Checking …
2023-10-24 17:57:52 : REQ   : firefoxpkgintl : ################## Start Installomator v. 10.5, date 2023-10-15
2023-10-24 17:57:52 : INFO  : firefoxpkgintl : ################## Version: 10.5
2023-10-24 17:57:52 : INFO  : firefoxpkgintl : ################## Date: 2023-10-15
2023-10-24 17:57:52 : INFO  : firefoxpkgintl : ################## firefoxpkgintl
2023-10-24 17:57:52 : INFO  : firefoxpkgintl : Installomator did not close any apps, so no need to reopen any apps.
2023-10-24 17:57:52 : ERROR : firefoxpkgintl : ERROR: unknown label firefoxpkgintl
2023-10-24 17:57:52 : REQ   : firefoxpkgintl : ################## End Installomator, exit code 1 
App Auto-Patch: 2023-10-24 17:57:52 - [ERROR] Error installing firefoxpkgintl. Exit code 0
```

**Comment by Apfelpom** [comment]
This error still occurs in version 2.0.0, when `ignoredLabels` contains `firefox*` and `requiredLabels` contains `firefoxpkg_intl` (same behavior when `firefoxpkg_intl` is specified as `optionalLabels`): the `firefoxpkg_intl` label is always accompanied by a non-existent `firefoxpkgintl`. In the logs, it first appears in the `Labels to install` after the discovery phase, it is also added to the list `Ignoring labels` and it also appears in the required and optional labels.
  
```
[NOTICE] Labels to install:  etrecheck firefoxpkg_intl firefoxpkgintl 
App Auto-Patch: 2023-12-20 11:03:56 - [NOTICE] Ignoring labels:  1password7 1password8 bbedit cloudya daisydisk defaultfolderx discord dymoconnectdesktop filemakerpro firefox firefox* firefox_da firefox_intl firefoxda firefoxdeveloperedition firefoxesr firefoxesr_intl firefoxesrintl firefoxintl firefoxpkg firefoxpkg_intl firefoxpkgintl googlechrome googlechrome* googlechromeenterprise googlechromepkg mactracker microsoft* microsoftautoupdate microsoftazuredatastudio microsoftazurestorageexplorer microsoftcompanyportal microsoftdefender microsoftedge microsoftexcel microsoftexcelreset microsoftlicenseremovaltool microsoftlicensereset microsoftoffice365 microsoftofficebusinesspro microsoftofficefactoryreset microsoftofficeremoval microsoftonedrive microsoftonedrive-deferred microsoftonedrive-rollingout microsoftonedrive-rollingoutdeferred microsoftonedrivedeferred microsoftonedrivereset microsoftonedriverollingout microsoftonedriverollingoutdeferred microsoftonenote microsoftonenotereset microsoftoutlook microsoftoutlook-monthly microsoftoutlookdataremoval microsoftoutlookmonthly microsoftoutlookreset microsoftpowerpoint microsoftpowerpointreset microsoftremotedesktop microsoftsharepointplugin microsoftskypeforbusiness microsoftskypeforbusinessremovaltool microsoftteams microsoftteamsnew microsoftteamsreset microsoftvisualstudiocode microsoftword microsoftwordreset packages suspiciouspackage swiftdialog viscosity 
[NOTICE] Required labels:  firefoxpkg_intl firefoxpkgintl 
[NOTICE] Optional Labels:  firefoxpkg_intl firefoxpkgintl
```

**Comment by robjschroeder** [comment]
Hey @Apfelpom , thanks for commenting on this issue. I realized what happened here, this will be fixed in `2.0.1` that should be released soon. I just tested on my locally saved version, it looks good.

**Comment by robjschroeder** [comment]
@Apfelpom can you test out `2.0.1`, it does have its own branch, but you're Firefox label issue should be fixed there

**Comment by robjschroeder** [comment]
@Apfelpom , this is resolved in version `2.0.1` or later. Check out #42 as the latest commit.
